osu.Game.Rulesets.Osu/Difficulty/OsuPerformanceCalculator.cs

```diff
@@ -262,6 +256,9 @@ private int calculateEffectiveMissCount()
             return Math.Max(countMiss, (int)Math.Floor(comboBasedMissCount));
         }
 
+        // Miss penalty assumes that a player will miss on the relatively hard parts of a map, not the easy parts, hence the strain count.
```

Comment:
Maybe something like "Miss penalty assumes that a player will miss on the hardest parts of a map, so we use amount of relatively difficult sections to adjust miss penalty to make it more punishing on maps with lower amount of hard sections."